docs/tutorials/clojure-guide.md

```diff
@@ -0,0 +1,225 @@
+
+You can find a function definition with `lsp-find-definition` or find all the references to that 
+symbol, variable or function with `lsp-find-references` or `lsp-ui-peek-find-references` if 
+[lsp-ui](https://emacs-lsp.github.io/lsp-ui/) is available. 
+
+![find-references](images/clojure-find-references.gif "find-references")
+
+<details>
+<summary>Doom emacs users</summary>
+
+You may need to remove the lookup handlers conflict from `cider` and/or `clj-refactor` if you want to use this LSP feature.
+
+```elisp
+(use-package! cider
+  :after clojure-mode
+  :config
+  (set-lookup-handlers! 'cider-mode nil))
+  
+(use-package! clj-refactor
+  :after clojure-mode
+  :config
+  (set-lookup-handlers! 'clj-refactor-mode nil))
+```
+
+</details>
+
+### Code lens
+
+LSP supports code lens that are texts that server may want to present in client IDE. `clojure-lsp` 
+use it for references count of a function/variable, they are disabled by default so you need to manually enable via:
+
+```elisp
+(setq lsp-lens-enable t)
+```
+
+![reference code-lens](images/clojure-code-lens.png "reference code-lens")
+
+### Call hierarchy
+
+This is a feature recently introduced in LSP 3.16, which servers should return the call hierarchy of a function/method.
+As of now, `clojure-lsp` supports only incoming call hierarchy:
+
+![incoming call-hierarchy](images/clojure-call-hierarchy.png "incoming call-hierarchy")
+
+It uses [lsp-treemacs](https://github.com/emacs-lsp/lsp-treemacs) which you should install to use this feature.
+
+### Formatting
+
+For LSP formatting, `clojure-lsp` use [cljfmt](https://github.com/weavejester/cljfmt) with support for all known `cljfmt` 
+settings.
+
+![formatting](images/clojure-formatting.gif "formatting")
+
+### Code actions
+
+One of the most important features considered by me is the **code actions** feature provided by the server. they are refactoring actions
+that may change your code after applied, `clojure-lsp` some of the refactorings from below are suggested by the server depending
+where user cursor is, for example `Add missing namespace` where `clojure-lsp` will suggest adding a missing require:
+  
+![add-missing-ns](images/clojure-add-missing-ns.gif "add-missing-ns")
+
+### Refactorings
+
+`clojure-lsp` also has a lot of refactorings, some of them can be automatically triggered by 
+code actions like said above and others manually by user, one example of one of the most 
+used by me is `clean-ns`, which will remove any unused requires and refers:
+
+![clean-ns](images/clojure-clean-ns.gif "clean-ns")
+
+Or `rename` which will rename the symbol at point and all usages/references:
+
+![rename](images/clojure-rename.gif "rename")
+
+For more info on all available refactorings, check [here](https://github.com/clojure-lsp/clojure-lsp#refactorings).
+
+### Semantic tokens (Experimental)
+
+Another feature recently introduced in LSP 3.16, is **semantic tokens**, which consists of the LSP server return to 
```

Comment:
Looking at the screenshot I couldn't quite understand how this is supposed to change the font-locking, so you might want to expand on this. I'm wondering if it interferes with CIDER's dynamic font-locking. (I assume it does)

Reply:
Yes, lsp-mode overrides cider font-lock